### PR TITLE
Reintroduce postgres9.5/postgis2.3 and avoided to remove python2.7 

### PR DESCRIPTION
**Pull request by luipir**
## Description
Some integrations test are configured to not use postgres on docker due to a more detailed configuration. so it's necessary to reintroduce a postgres server that can be configured and run in the test environment.

the docker where tests run has been changed
Adding:
Postgres 9.5
Postgis 2.3
Avoid removing of:
python2.7
that, for some ubuntu reason fail raising unmet dependencies (when the dependency is available)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by luipir** [comment]
need to update tests to use a correct setting of  QGIS_POSTGRES_EXECUTABLE_PATH default is pointing to postgres 9.4 instead of the installed 9.5

**Comment by m-kuhn** [comment]
Looks good to install postgres.
Why does python 2.7 need to be reintroduced though?

**Comment by luipir** [comment]
it's not reintroduced explicitally, some installed module seems need it as dependency and can be removed :(

**Comment by m-kuhn** [comment]
Installed modules of postgres? Do you have an idea which ones?

**Comment by luipir** [comment]
@m-kuhn looking for it

File: .ci/travis/linux/docker-build-test.sh [hunk 1]
```diff
@@ -81,6 +81,14 @@ pushd /root/QGIS > /dev/null
 /root/QGIS/tests/testdata/provider/testdata_pg.sh
 popd > /dev/null # /root/QGIS
 
+##########################################################
+# Set Postgres server location for that tests that require
+# setting up a custom server and not use a standard 
+# dockerised one
```
Comment (m-kuhn) [inline .ci/travis/linux/docker-build-test.sh]: this one is also dockerized, we're running in a docker container here, the other one just runs in a dedicated docker ;)

Comment (luipir) [reply .ci/travis/linux/docker-build-test.sh]: well.. sure I'll fix the comment

**Comment by m-kuhn** [comment]
Thanks for taking care of this @luipir , will be great to have these tests re-enabled.

**Comment by luipir** [comment]
it is reintroduced installing postgresql-9.5-postgis-2.3 !

**Comment by luipir** [comment]
or better... any remove will fail due to libopenscenegraph100v5 unmet dependency
libopenscenegraph100v5 is introduced by libsfcgal1 introduced by postgresql-9.5-postgis-2.3

**Comment by luipir** [comment]
rebased to to get build fixes upstream

**Comment by m-kuhn** [comment]
Does this mean, python 2.7 is introduced as dependency by postgis (through sfcgal / osg)?
Sounds good to me in this case.
Sorry, I've lost you a bit on the latest comments.

**Comment by luipir** [comment]
"Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming."
I suppose its' because we are using ubuntugis-unstable... any reason to use unstable @3nids ?

**Comment by m-kuhn** [comment]
See https://github.com/qgis/QGIS/pull/5550#issuecomment-342486353

**Comment by luipir** [comment]
inside the docker instance: 
root@0bb334287ae5:/# apt-cache search postgis
apt-cache search postgis
liblwgeom-2.2-5 - PostGIS "Lightweight Geometry" library
liblwgeom-dev - PostGIS "Lightweight Geometry" library - Development files
libpostgis-java - Geographic objects support for PostgreSQL -- JDBC support
postgis - Geographic objects support for PostgreSQL
postgis-doc - Geographic objects support for PostgreSQL -- documentation
postgresql-9.5-postgis-2.2 - Geographic objects support for PostgreSQL 9.5
postgis-gui - Geographic objects support for PostgreSQL -- GUI programs
postgresql-9.5-postgis-scripts - transitional dummy package
postgresql-9.5-pgrouting - Routing functionality support for PostgreSQL/PostGIS
postgresql-9.5-pgrouting-doc - Routing functionality support for PostgreSQL/PostGIS (Documentation)
liblwgeom-2.3-0 - PostGIS "Lightweight Geometry" library
postgresql-9.5-postgis-2.3 - Geographic objects support for PostgreSQL 9.5
postgresql-9.5-postgis-2.3-scripts - Geographic objects support for PostgreSQL 9.5 -- scripts
postgresql-9.5-pgrouting-scripts - Routing functionality support for PostgreSQL/PostGIS - scripts

no other postgis can be installed. postgresql-9.5-postgis-2.2  link libsfcgal1 too => no more solution are available other than avoid the error.
I tried various apt-get combination to fix the error, but it is introduced because we are using unstable. Probably the error will disappear during evolution of unstable packages

**Comment by luipir** [comment]
@m-kuhn wait for merge of https://github.com/qgis/QGIS/commit/ac574e231ac7e7c1cbcb827f59ae8a2ee5e2739a that shoudl fix the build and pass (I hope) the tests

**Comment by luipir** [comment]
@m-kuhn postgres test has been updated but seems the docker container is still the old one without postgres installed. how update the contained based on the new dockerfile?

**Comment by luipir** [comment]
@m-kuhn Boundless fired most of Desktop team and blocked any GFOSS contribution, so in the meantime I can't work on it in the next days. If someone take care of auth PKI tests now it's all delegated to community.
Probably I'll come back on this after a time to find new incomes.
Cheers

**Comment by jef-n** [comment]
@luipir is there more information on this?

**Comment by luipir** [comment]
@jef-n about boundless or the PR?

**Comment by luipir** [comment]
moved to #6664

**Pull request closed without merging**